app/v2/coinswap/internal/keeper/swap.go

```diff
@@ -73,21 +71,29 @@ func (k Keeper) IsDoubleSwap(ctx sdk.Context, denom1, denom2 string) bool {
 	return denom1 != sdk.IrisAtto && denom2 != sdk.IrisAtto
 }
 
-// GetExchangeName returns the ModuleAccount name for the provided denominations.
-// The module name is in the format of 'swap:denom:denom' where the denominations
-// are sorted alphabetically.
-func (k Keeper) GetExchangeName(denom1, denom2 string) (string, sdk.Error) {
-	switch strings.Compare(denom1, denom2) {
-	case -1:
-		return "swap:" + denom1 + ":" + denom2, nil
-	case 1:
-		return "swap:" + denom2 + ":" + denom1, nil
-	default:
-		return "", types.ErrEqualDenom("denomnations for forming module name are equal")
+// GetReservePoolName returns the reserve pool name for the provided denominations.
+// The reserve pool name is in the format of 's-denom' which the denomination
+// is not iris-atto.
+func (k Keeper) GetReservePoolName(denom1, denom2 string) (string, sdk.Error) {
+	if denom1 == denom2 {
+		return "", types.ErrEqualDenom("denomnations for forming reserve pool name are equal")
+	}
+
+	if denom1 != sdk.IrisAtto && denom2 != sdk.IrisAtto {
+		return "", types.ErrIllegalDenom(fmt.Sprintf("illegal denomnations for forming reserve pool name, must have one native denom: %s", sdk.IrisAtto))
```

Comment:
Now, not only one native token. native denom -> iris-atto?

Reply:
`fmt.Sprintf("illegal denomnations for forming reserve pool name, must have one native denom: %s", sdk.IrisAtto)` will be formatted to `illegal denomnations for forming reserve pool name, must have one native denom: iris-atto`

Reply:
my mistake.